LeaderboardBackend/Services/Impl/RunService.cs

```diff
@@ -72,51 +68,53 @@ Page page
             return new NotFound();
         }
 
-        string direction = cat.SortDirection switch
-        {
-            SortDirection.Ascending => "ASC",
-            SortDirection.Descending => "DESC",
-            _ => throw new InvalidEnumArgumentException(),
-        };
-
-        // The linter check we disable here checks for calls to .FromSqlRaw.
-        // We need to call .FromSqlRaw to be able to pass `direction` into ORDER BY.
-#pragma warning disable EF1002
-        IQueryable<Run> initQuery = applicationContext.Runs.FromSqlRaw($"""
-        SELECT *
-        FROM (
-            SELECT r.*, ROW_NUMBER() OVER (PARTITION BY r.user_id ORDER BY r.time_or_score {direction}, r.played_on, r.created_at, r.id) as row_number
-            FROM runs as r
-            WHERE r.category_id = {id} AND r.deleted_at IS NULL
-        ) as t
-        WHERE t.row_number = 1
-        """);
-#pragma warning restore EF1002
-
-        long count = await initQuery.LongCountAsync();
-
-        IIncludableQueryable<Run, User> unsorted = initQuery
-            .Include(r => r.User);
-
-        IOrderedQueryable<Run> runsOrdered;
-        if (cat.SortDirection is SortDirection.Ascending)
+        bool asc = cat.SortDirection == SortDirection.Ascending;
+
+        // Query the best run from each user.
+
+        IQueryable<RankedRun> query = applicationContext.Runs
+            .Include(r => r.Category)
+            .Include(r => r.User)
+            .Where(r => r.CategoryId == id && r.DeletedAt == null)
+            .Where(r => EF.Functions.RowNumber((
+                asc ?
+                EF.Functions.Over().PartitionBy(r.UserId).OrderBy(r.TimeOrScore) :
+                EF.Functions.Over().PartitionBy(r.UserId).OrderByDescending(r.TimeOrScore)
+            ).ThenBy(r.PlayedOn).ThenBy(r.CreatedAt)) == 1L)
+
+        // Assign each run a rank relative to the other users' best runs and count them up.
+
+            .Select(r => new RankedRun
+            {
+                Rank = EF.Functions.Rank(
+                    asc ?
+                    EF.Functions.Over().OrderBy(r.TimeOrScore) :
+                    EF.Functions.Over().OrderByDescending(r.TimeOrScore)),
+                Run = r,
+                Count = EF.Functions.Count<long>(EF.Functions.Over())
```

Comment:
What would Count's value be, for example? Also how is it being used? I don't see it being used in e.g. the ViewModel. What's it for?